include/fmt/chrono.h

```diff
@@ -451,24 +451,16 @@ FMT_END_DETAIL_NAMESPACE
 template <typename Char, typename Duration>
 struct formatter<std::chrono::time_point<std::chrono::system_clock, Duration>,
                  Char> : formatter<std::tm, Char> {
-  FMT_CONSTEXPR FMT_INLINE void generate_default_spec(char) {
-    this->specs = {"%Y-%m-%d %H:%M:%S", 17};
-  }
-  FMT_CONSTEXPR FMT_INLINE void generate_default_spec(wchar_t) {
-    this->specs = {L"%Y-%m-%d %H:%M:%S", 17};
-  }
-
   template <typename ParseContext>
   FMT_CONSTEXPR auto parse(ParseContext& ctx) -> decltype(ctx.begin()) {
     auto it = ctx.begin();
     if (it != ctx.end() && *it == ':') ++it;
     auto end = it;
     while (end != ctx.end() && *end != '}') ++end;
-    if (end == it) {
-      generate_default_spec(typename ParseContext::char_type{});
-    } else {
+    if (end == it)
+      this->specs = {default_spec, 17};
```

Comment:
Maybe replace constant ```17``` to ```std::end(default_spec) - std::begin(default_spec)``` ?
```std::begin``` and ```std::end``` functions is ```constexpr``` since C++14.

Reply:
My mistake.
Correct code:
```detail::to_unsigned(std::end(default_spec) - std::begin(default_spec))```

Reply:
I don't quite see the point here. Since we are using an array, the result should be certain at compile time, also this sentence is too long. Maybe we can use a comment to explain?

Reply:
I think this is a design choice, not a technical issue. I will leave the choice to @vitaut .

Reply:
I suggest replacing 17 with `sizeof(default_spec) / sizeof(Char)`.